### ISSUE TITLE
[java] SimplifyBooleanReturns: False negative when if-return is wrapped inside unnecessary anonymous block

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6519]
**Affects PMD Version:** 7.22.0

**Rule:**[SimplifyBooleanReturns](https://pmd.github.io/pmd/pmd_rules_java_design.html#simplifybooleanreturns)

**Description:**
The SimplifyBooleanReturns rule fails to trigger on the classic ```if-return``` boolean pattern when it is wrapped inside an unnecessary anonymous block ```{}```.

This is a clear false negative — both patterns are semantically identical and can be safely replaced by a single ```return``` statement, yet only the direct case is flagged.

**Code Sample demonstrating the issue:**

```java
public class a{
    public boolean TP(Object o) {
        if(obj == null) return false;   // TP: correctly reported
        return true;
    }
    public boolean FN(Object o) {
        {
            if(obj == null) return false;   // FN: should report but doesn't
        }
        return true;
    }
}
```

**Expected outcome:**
PMD should report SimplifyBooleanReturns for the if statement in both ```TP()``` and ```FN()``` with the message:
"This if statement can be replaced by return ```!{condition}```;"
But currently it only reports the direct case (TP), completely **missing** the extra block case (FN).

**Running PMD through:** *CLI*

**Comment by adangel** [comment]
Closing this issue in favor of #6639 which introduces a new rule to detect unnecessary blocks.
